protocol/identity/alias/generate.go

```diff
@@ -32,16 +30,31 @@ func GenerateFromPublicKey(publicKey *ecdsa.PublicKey) string {
 // GenerateFromPublicKeyString returns the 3 words name given a public key
 // prefixed with 0x
 func GenerateFromPublicKeyString(publicKeyString string) (string, error) {
-	pk := strings.TrimPrefix(publicKeyString, "0x")
-	publicKeyBytes, err := hex.DecodeString(pk)
-	if err != nil {
-		return "", err
+	var fixedPublicKey string
+	if len(publicKeyString) == multiformat.LegacyKeyLength {
+		fixedPublicKey = publicKeyString
+	} else {
+		// Attempt to fix a maybe public key without the 0x prefix
+		if len(publicKeyString) >= 2 && publicKeyString[:2] != "0x" {
+			fixedPublicKey = "0x" + publicKeyString
+		} else {
+			return "", fmt.Errorf("invalid length of public key")
+		}
 	}
 
-	publicKey, err := crypto.UnmarshalPubkey(publicKeyBytes)
+	compressedKey, err := multiformat.SerializeLegacyKey(fixedPublicKey)
 	if err != nil {
 		return "", err
 	}
+	return ShortenedCompressedKey(compressedKey), nil
+}
 
-	return GenerateFromPublicKey(publicKey), nil
+func ShortenedCompressedKey(compressedKey string) string {
+	if len(compressedKey) > 12 {
+		prefix := compressedKey[0:8]
+		suffix := compressedKey[len(compressedKey)-5:]
+		return prefix + "…" + suffix
+	} else {
+		return ""
```

Comment:
We love the early return pattern 🙂 
```suggestion
	if len(compressedKey) <= 12 {
	    return ""
    }
    prefix := compressedKey[0:8]
    suffix := compressedKey[len(compressedKey)-5:]
    return prefix + "…" + suffix
```